docscan/session/create/filter/required_supplementary_document.go

```diff
@@ -0,0 +1,84 @@
+package filter
+
+import (
+	"encoding/json"
+
+	"github.com/getyoti/yoti-go-sdk/v3/docscan/session/create/objective"
+)
+
+// RequiredSupplementaryDocument details a required supplementary document
+type RequiredSupplementaryDocument struct {
+	Filter        RequestedDocumentFilter
+	DocumentTypes *[]string
+	CountryCodes  *[]string
+	Objective     objective.Objective
+}
+
+// Type returns the type of the supplementary document
+func (r *RequiredSupplementaryDocument) Type() string {
+	return supplementaryDocument
+}
+
+// MarshalJSON returns the JSON encoding
+func (r *RequiredSupplementaryDocument) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Type          string                  `json:"type"`
+		Filter        RequestedDocumentFilter `json:"filter,omitempty"`
+		CountryCodes  *[]string               `json:"country_codes,omitempty"`
+		DocumentTypes *[]string               `json:"document_types,omitempty"`
+		Objective     objective.Objective     `json:"objective,omitempty"`
+	}{
+		Type:          r.Type(),
+		Filter:        r.Filter,
+		CountryCodes:  r.CountryCodes,
+		DocumentTypes: r.DocumentTypes,
+		Objective:     r.Objective,
+	})
+}
+
+// NewRequiredSupplementaryDocumentBuilder creates a new RequiredSupplementaryDocumentBuilder
+func NewRequiredSupplementaryDocumentBuilder() *RequiredSupplementaryDocumentBuilder {
+	return &RequiredSupplementaryDocumentBuilder{}
+}
+
+// RequiredSupplementaryDocumentBuilder builds a RequiredSupplementaryDocument
+type RequiredSupplementaryDocumentBuilder struct {
+	filter        RequestedDocumentFilter
+	documentTypes *[]string
```

Comment:
Presume we're using pointers here, since we want them to be omitted from the payload if they're not specified?

Reply:
Yes, we could just omit empty arrays, but that would be different to other SDKs. I can change if that's preferable?

The builder methods don't use a pointer, e.g.
```go
func (r *RequiredSupplementaryDocumentBuilder) WithCountryCodes(countryCodes []string) 
```

Reply:
Omitting empty arrays in 6ae2e0f (no longer using a pointer)